sdk/tests/conformance/extensions/ext-disjoint-timer-query.html

```diff
@@ -123,10 +134,65 @@
     wtu.glErrorShouldBe(gl, gl.NO_ERROR, "getParameter disjoint calls should work.");
 }
 
+function runElapsedTimeTest() {
+    debug("");
+    debug("Testing elapsed time query");
+
+    elapsed_query = ext.createQueryEXT();
+    ext.beginQueryEXT(ext.TIME_ELAPSED_EXT, elapsed_query);
+    gl.clearColor(0, 0, 1, 1);
+    gl.clear(gl.COLOR_BUFFER_BIT);
+    ext.endQueryEXT(ext.TIME_ELAPSED_EXT);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Time elapsed query should have no errors");
+}
+
+function runTimeStampTest() {
+    debug("");
+    debug("Testing timestamp query");
+
+    timestamp_query1 = ext.createQueryEXT();
+    timestamp_query2 = ext.createQueryEXT();
+    ext.queryCounterEXT(timestamp_query1, ext.TIMESTAMP_EXT);
+    gl.clearColor(1, 0, 0, 1);
+    gl.clear(gl.COLOR_BUFFER_BIT);
+    ext.queryCounterEXT(timestamp_query2, ext.TIMESTAMP_EXT);
+    wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Timestamp queries should have no errors");
+}
+
+function checkQueryResults() {
+    debug("");
+    debug("Testing query results");
+
+    // Make sure queries are available.
+    gl.finish();
```

Comment:
In Chrome at least gl.finish actually executes a flush. The only way to actually block and finish it to call gl.readPixels. You could do that here or (more elegantly IMO) check for QUERY_RESULT_AVAILABLE in a rAF loop, timing out if you don't see a success in a reasonable number of iterations/milliseconds. Otherwise this test looks good!